alpha/action/reversemigrate.go

```diff
@@ -0,0 +1,201 @@
+						return fmt.Errorf("insert tail channel entry %q for package %q, channel %q: %v", cur.Name, pkg.Name, ch.Name, err)
+					}
+				}
+				depth += 1
+				channelEntryID += 1
+				cur = next
+			}
+
+			cur = head
+			for cur != nil {
+				for _, skip := range cur.Skips {
+					if _, err := tx.Exec("INSERT INTO channel_entry (entry_id, channel_name, package_name, operatorbundle_name, depth) VALUES (?, ?, ?, ?, ?)", channelEntryID, ch.Name, pkg.Name, skip, depth); err != nil {
+						return fmt.Errorf("insert 'skip to' channel entry %q for package %q, channel %q: %v", skip, pkg.Name, ch.Name, err)
+					}
+					channelEntryID += 1
+					if _, err := tx.Exec("INSERT INTO channel_entry (entry_id, channel_name, package_name, operatorbundle_name, replaces, depth) VALUES (?, ?, ?, ?, ?, ?)", channelEntryID, ch.Name, pkg.Name, cur.Name, channelEntryID-1, depth); err != nil {
+						return fmt.Errorf("insert 'skip from' channel entry %q for package %q, channel %q: %v", cur.Name, pkg.Name, ch.Name, err)
+					}
+					channelEntryID += 1
+				}
+				cur = ch.Bundles[cur.Replaces]
+			}
+
+			for _, b := range ch.Bundles {
+				if existing, ok := addedBundles[b.Name]; ok {
+					if !equivalentBundles(*existing, *b) {
+						return fmt.Errorf("cannot produce sqlite-equivalent of FBC: found unsupported differences between channels for bundle %q", b.Name)
+					}
+
+					continue
+				}
+				addedBundles[b.Name] = b
+
+				for _, ri := range b.RelatedImages {
+					if _, err := tx.Exec("INSERT INTO related_image (image, operatorbundle_name) VALUES (?, ?)", ri.Image, b.Name); err != nil {
+						return fmt.Errorf("insert related image %q for package %q, channel %q, bundle %q: %v", ri.Image, pkg.Name, ch.Name, b.Name, err)
+					}
+				}
+				for _, p := range b.Properties {
+					if p.Type == "olm.bundle.object" {
+						continue
+					}
+					if _, err := tx.Exec("INSERT INTO properties (type, value, operatorbundle_name, operatorbundle_version, operatorbundle_path) VALUES (?, ?, ?, ?, ?)", p.Type, string(p.Value), b.Name, b.Version, b.Image); err != nil {
+						return fmt.Errorf("insert property %q for package %q, channel %q, bundle %q: %v", p.Type, pkg.Name, ch.Name, b.Name, err)
+					}
+					if p.Type == "olm.deprecated" {
+						if _, err := tx.Exec("INSERT INTO deprecated (operatorbundle_name) VALUES (?)", b.Name); err != nil {
+							return fmt.Errorf("insert deprecation for package %q, channel %q, bundle %q: %v", pkg.Name, ch.Name, b.Name, err)
+						}
+					}
+				}
+				// TODO QUESTION: Do we need to ever insert a substitutesFor value?
+				//    I think, no. FBC doesn't use the subsFor concept in its API.
+				//    The expectation is that the FBC already contains all of the
+				//    graph edges and changes to account for substitutions, and that
+				//    any further substitutions will be performed via the FBC and
+				//    then another reverse-migrate will occur to convert to sqlite.
```

Comment:
or we need to change the way the substitutesfor logic works in sqlite because I believe it relies on this field being in the db for substitutions of substitutions